packages/reactotron-app/App/Dialogs/FilterTimelineDialog.js

```diff
@@ -0,0 +1,246 @@
+import React, { Component } from 'react'
+import ReactDOM from 'react-dom'
+import { ModalPortal, ModalBackground, ModalDialog } from 'react-modal-dialog'
+import { inject, observer } from 'mobx-react'
+import AppStyles from '../Theme/AppStyles'
+import Colors from '../Theme/Colors'
+import Checkbox from '../Shared/Checkbox'
+
+// Move this to a better place?
+const FILTER_OPTIONS = [
+  {
+    name: 'General',
+    items: [
+      {
+        value: 'client.intro',
+        text: 'Connected'
+      },
+      {
+        value: 'benchmark.report',
+        text: 'Benchmark'
+      },
+      {
+        value: 'log',
+        text: 'Log Messages'
+      },
+      {
+        value: 'image',
+        text: 'Images'
+      },
+      {
+        value: 'display',
+        text: 'Display'
+      }
+    ]
+  },
+  {
+    name: 'API',
+    items: [
+      {
+        value: 'api.response',
+        text: 'API Responses'
+      }
+    ]
+  }
+  {
+    name: 'Redux',
+    items: [
+      {
+        value: 'state.action.complete',
+        text: 'Action'
+      },
+      {
+        value: 'saga.task.complete',
+        text: 'Saga'
+      },
+      {
+        value: 'state.values.response',
+        text: 'State Values'
+      },
+      {
+        value: 'state.values.response',
+        text: 'State Keys'
+      },
+      {
+        value: 'state.values.change',
+        text: 'State Values Change'
+      }
+    ]
+  }
+]
+
+const DIALOG_TITLE = 'Filter'
+
+const Styles = {
+  dialog: {
+    borderRadius: 4,
+    padding: 4,
+    width: 450,
+    backgroundColor: Colors.background,
+    color: Colors.foreground
+  },
+  examples: {
+  },
+  example: {
+    padding: 0,
+    margin: '0 0 0 40px',
+    color: Colors.bold
+  },
+  container: {
+    ...AppStyles.Layout.vbox
+  },
+  keystrokes: {
+    ...AppStyles.Layout.hbox,
+    alignSelf: 'center',
+    paddingTop: 10,
+    paddingBottom: 20,
+    fontSize: 13
+  },
+  hotkey: {
+    padding: '0 10px'
+  },
+  keystroke: {
+    backgroundColor: Colors.backgroundHighlight,
+    color: Colors.foreground,
+    padding: '4px 8px',
+    borderRadius: 4
+  },
+  header: {
+    ...AppStyles.Layout.vbox,
+    padding: '1em 2em 0em'
+  },
+  body: {
+    ...AppStyles.Layout.vbox,
+    padding: '1em 2em 4em'
+  },
+  title: {
+    margin: 0,
+    padding: 0,
+    textAlign: 'left',
+    fontWeight: 'normal',
+    fontSize: 24,
+    color: Colors.heading
+  },
+  subtitle: {
+    color: Colors.foreground,
+    textAlign: 'left',
+    padding: 0,
+    margin: 0
+  },
+  fieldLabel: {
+    color: Colors.heading,
+    fontSize: 13,
+    textTransform: 'uppercase'
+  },
+  textField: {
+    borderTop: 0,
+    borderLeft: 0,
+    borderRight: 0,
+    borderBottom: `1px solid ${Colors.line}`,
+    fontSize: 23,
+    color: Colors.foregroundLight,
+    lineHeight: '40px',
+    backgroundColor: 'inherit'
+  }
+}
+
+const INSTRUCTIONS = <div>
```

Comment:
Still have to figure out if we need instructions on this dialog and what to have them say if so